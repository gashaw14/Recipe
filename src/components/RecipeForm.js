import React from 'react';
import IngredientList from './IngredientList';

export default class RecipeForm extends React.Component{

constructor(props){
super(props);
this.state={
    menuTitle: props.recipe ? props.recipe.menuTitle:'',
    steps: props.recipe ? props.recipe.steps:'',
    ingredient: props.recipe ? props.recipe.ingredient:'',
    checkbox:props.recipe ? props.recipe.checkbox: false,
    error:''
}

}
onMenuTitleChange=(e)=>{
 const menuTitle = e.target.value
 this.setState(()=>({menuTitle}))
}

onStepsChange =(e)=>{
 const steps = e.target.value
 this.setState(()=>({steps}))
}
onIngredientChange=(e)=>{
const ingredient = e.target.value;
this.setState(()=>({ingredient}));
}

handleSubmit= (e)=>{
e.preventDefault()
if(!this.state.menuTitle ){
this.setState(()=>({error:'please enter menu title'}))
} else{
    this.setState(()=>({errors:''}))
    this.props.onSubmit({
        menuTitle: this.state.menuTitle,
        steps: this.state.steps,
        ingredient: this.state.ingredient,
        checkbox: !this.state.checkbox
    })
}
}

render(){
    return(
    <div className='form'>
      {this.state.error && <p>{this.state.error}</p>}
     <form onSubmit={this.handleSubmit}>
         <input
         type='text'
         placeholder='menu name'
         value= {this.state.menuTitle}
         onChange ={this.onMenuTitleChange}
         ></input>
         <textarea
         type='text'
         placeholder='cooking steps'
         value={this.state.steps}
         onChange={this.onStepsChange}
         ></textarea>
        <h1>Ingredients</h1>
        <IngredientList />
        <input 
        type='text' 
        placeholder='add ingredient'
        value={this.state.ingredient}
        onChange={this.onIngredientChange}
        ></input>
        <button >Add Ingredient</button>
     </form>   
     
    </div>
    )
}
}