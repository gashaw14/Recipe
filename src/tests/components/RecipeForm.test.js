import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import RecipeForm from '../../components/RecipeForm';

const recipes =[{
    id:'1',
    menuTitle:'VEGI', 
    steps:'ttt',
    ingredient:'aaa',
    checkbox:false

},{
   id:'2',
    menuTitle:'meat', 
    steps:'cook',
    ingredient:'yyyy',
    checkbox:false
}
]

test('default recipe form',()=>{
    const wrapper = shallow(<RecipeForm />);
    expect(toJSON(wrapper)).toMatchSnapshot()
})

test('test recipe form with data',()=>{
    const wrapper = shallow(<RecipeForm recipe={recipes[1]} />)
    expect(toJSON(wrapper)).toMatchSnapshot()

})

test('test invalid recipe form submision',()=>{
    const wrapper = shallow(<RecipeForm />);
    expect(toJSON(wrapper)).toMatchSnapshot()
    wrapper.find('form').simulate('submit',{
        preventDefault:()=>{}
    })
   expect(wrapper.state('error').length).toBeGreaterThan(0)
   expect(toJSON(wrapper)).toMatchSnapshot()
})

test('set menu input change recipe form',()=>{
    const value = 'new value'
    const wrapper = shallow(<RecipeForm />);
    wrapper.find('input').at(0).simulate('change',{
        target:{value}
    })
    expect(wrapper.state('menuTitle')).toBe(value)
    expect(toJSON(wrapper)).toMatchSnapshot()
})
test('set step input change recipe form',()=>{
    const value = 'here are steps for recipe'
    const wrapper = shallow(<RecipeForm />);
    wrapper.find('textarea').at(0).simulate('change',{
        target:{value}
    })
    expect(wrapper.state('steps')).toBe(value)
    expect(toJSON(wrapper)).toMatchSnapshot()
})
test('test on submit form submission',()=>{
    const onSubmitSpy =  jest.fn();
    const wrapper = shallow(<RecipeForm recipe={recipes[1]} onSubmit={onSubmitSpy}/>)
    wrapper.find('form').simulate('submit',{
        preventDefault:()=>{}
    })
    expect(wrapper.state('error')).toBe('');
    expect(onSubmitSpy).toHaveBeenCalledWith({
        menuTitle: recipes[1].menuTitle,
        steps: recipes[1].steps,
        ingredient: recipes[1].ingredient,
        checkbox: !recipes[1].checkbox
    })
})