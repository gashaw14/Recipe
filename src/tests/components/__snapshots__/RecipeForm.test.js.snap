// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default recipe form 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="menu name"
      type="text"
      value=""
    />
    <textarea
      onChange={[Function]}
      placeholder="cooking steps"
      type="text"
      value=""
    />
    <h1>
      Ingredients
    </h1>
    <Connect(IngredientList) />
    <input
      onChange={[Function]}
      placeholder="add ingredient"
      type="text"
      value=""
    />
    <button>
      Add Ingredient
    </button>
  </form>
</div>
`;

exports[`set menu input change recipe form 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="menu name"
      type="text"
      value="new value"
    />
    <textarea
      onChange={[Function]}
      placeholder="cooking steps"
      type="text"
      value=""
    />
    <h1>
      Ingredients
    </h1>
    <Connect(IngredientList) />
    <input
      onChange={[Function]}
      placeholder="add ingredient"
      type="text"
      value=""
    />
    <button>
      Add Ingredient
    </button>
  </form>
</div>
`;

exports[`set step input change recipe form 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="menu name"
      type="text"
      value=""
    />
    <textarea
      onChange={[Function]}
      placeholder="cooking steps"
      type="text"
      value="here are steps for recipe"
    />
    <h1>
      Ingredients
    </h1>
    <Connect(IngredientList) />
    <input
      onChange={[Function]}
      placeholder="add ingredient"
      type="text"
      value=""
    />
    <button>
      Add Ingredient
    </button>
  </form>
</div>
`;

exports[`test invalid recipe form submision 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="menu name"
      type="text"
      value=""
    />
    <textarea
      onChange={[Function]}
      placeholder="cooking steps"
      type="text"
      value=""
    />
    <h1>
      Ingredients
    </h1>
    <Connect(IngredientList) />
    <input
      onChange={[Function]}
      placeholder="add ingredient"
      type="text"
      value=""
    />
    <button>
      Add Ingredient
    </button>
  </form>
</div>
`;

exports[`test invalid recipe form submision 2`] = `
<div>
  <p>
    please enter menu title
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="menu name"
      type="text"
      value=""
    />
    <textarea
      onChange={[Function]}
      placeholder="cooking steps"
      type="text"
      value=""
    />
    <h1>
      Ingredients
    </h1>
    <Connect(IngredientList) />
    <input
      onChange={[Function]}
      placeholder="add ingredient"
      type="text"
      value=""
    />
    <button>
      Add Ingredient
    </button>
  </form>
</div>
`;

exports[`test recipe form with data 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="menu name"
      type="text"
      value="meat"
    />
    <textarea
      onChange={[Function]}
      placeholder="cooking steps"
      type="text"
      value="cook"
    />
    <h1>
      Ingredients
    </h1>
    <Connect(IngredientList) />
    <input
      onChange={[Function]}
      placeholder="add ingredient"
      type="text"
      value="yyyy"
    />
    <button>
      Add Ingredient
    </button>
  </form>
</div>
`;
